### PR TITLE
DAOS-5735 - All pool bad_evict tests failed on weekly due to DaosApiError: No existing UUID on pool.destroy

### PR DESCRIPTION
**Pull request by cibervicho**
DAOS-5735 - All pool bad_evict tests failed on weekly due to DaosApiError: No existing UUID on pool.destroy

Different updates have to be done to the test in order to remove the dependency of the
DaosPool.create() method, since when it was removed from pydaos this test started
to fail.
Now the test uses the DmgCommand for pool creation.

I also removed from the yaml file the different scenarios of pool create because
the pool create operation with dmg accepts any DAOS system name passing it
with the --sys flag/option.

### CODE REVIEW

**Comment by cibervicho** [comment]
Test passed. Adding reviewers.

File: src/tests/ftest/pool/bad_evict.py [hunk 1]
```diff
@@ -78,41 +65,35 @@ def test_evict(self):
...
                 saveduuid = (ctypes.c_ubyte * 16)(0)
                 for item in range(0, len(saveduuid)):
-                    saveduuid[item] = pool.uuid[item]
-                pool.uuid[4] = 244
+                    saveduuid[item] = pool.pool.uuid[item]
+                pool.pool.uuid[4] = 244
 
-            pool.evict()
+            # evict the pool
+            self.get_dmg_command().pool_evict(pool=pool.pool.get_uuid_str())
```
Comment (berserk-fury) [inline src/tests/ftest/pool/bad_evict.py:91]: Can we use the evict method?

`pool.pool.evict()`

Comment (cibervicho) [reply src/tests/ftest/pool/bad_evict.py]: Sure. Done.

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by berserk-fury — APPROVED** [review]

**Comment by cibervicho** [comment]
Test passing with requested changes.
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3916/9/artifact/Functional/job-2020-11-23T19.31-97a4bcf-pool-bad_evict/job.log

**Pull request merged**